### DISCUSSION TITLE
Not able to see Google Pay Button when testEnv set to true

### DISCUSSION THREAD

**Discussion by spaquis** [flutter-stripe/flutter_stripe#1541] [Q&A]
**Describe the bug**
Hi,
I'm trying to integrate the Google Pay button inside Payment Sheet but i can't succeed to display the Google pay button on a real device when testEnv property is set to true (Apple Pay works well) : 

` googlePay: const PaymentSheetGooglePay(
            merchantCountryCode: 'IE',
            currencyCode: 'EUR',
            testEnv: true,
          ),`

If i set testEvent to false, i can see the Google Pay button but when i click on it i have an error [Google Pay OR_BIBED_11] which is logic because my test environment is linked to stripe test mode. 

**To Reproduce**
Just add the code above inside Stripe.instance.initPaymentSheet after setup correctly Google Pay (Metadata in manifest / stripe android Sdk). Then launch the payment sheet on Android => the google pay button didn't appear

**Expected behavior**
The google pay button should appear with testEnv: true

**Smartphone / tablet**
 - Device: Samsung S21
 - OS: Android 13
 - Package version: 9.4.0
 -  - Flutter version 3.10.6

**Additional context**
As i said above, the button is displayed when testEnv is false (should demonstrate that the configuration is OK)

Thanks for your help

**Comment by jonasbark** [comment]
https://stripe.com/docs/google-pay

Double-check your configuration, e.g. merchant ID and similar. It's always a configuration issue.

**Reply by spaquis** [reply]
Hi @jonasbark ,
Thanks for your comment the merchant ID is only need for Apple Pay or I missed something ?

**Reply by Kanwarpalsingh1986** [reply]
Same issue

**Reply by remonh87** [reply]
did you add the following to your manifest?

```xml
<meta-data
                android:name="com.google.android.gms.wallet.api.enabled"
                android:value="true" />
```

**Reply by Kanwarpalsingh1986** [reply]
yes added that already, didnt worked

**Reply by iEquos** [reply]
I have exactly the same issue, where I only see the button with testEnv set to false and then get the OR_BIBED_11 error if clicked (also using stripe test key).
But I'm using the @stripe/stripe-react-native package in a react native app.

So maybe there's an issue in both packages?

**Reply by zeal-saloni-s** [reply]
I'm having the same issue with flutter

**Reply by remonh87** [reply]
That error is coming from Google pay itself. I would check their documentation. Keep in mind that you need to be accepted by Google if you use a production key

**Comment by captnCC** [comment]
@spaquis The fix is to enter a valid credit cards to your google pay wallet. This is needed needed especially for the testMode to work. Additionally you should enter yourself to the group for test cards: https://developers.google.com/pay/api/android/guides/resources/test-card-suite

After this steps, the "Pay with GPay" button should appear and you can do a test payment with the provided test cards

**Reply by empiteranga** [reply]
@captnCC Is it necessary to add a valid credit card to Google Wallet? In my region, it doesn't support adding a valid credit card. In that case, how do I test Google Payment? I need to get a screenshot of the Google Payment sheet from my app to submit for the Google Pay review process. I'm stuck here. Can you please advise me on how to resolve this? When I go with test mode set to false and live keys, the Google Pay button is displayed, but I can't continue the payment because Google Pay doesn't approve my account

**Reply by nayanbabariya** [reply]
@captnCC I am not able to see the `Google Pay` button in any mode (live or test), I've completely followed the guide mentioned in this link => https://docs.page/flutter-stripe/flutter_stripe/sheet#7-optional---enable-google-pay-only-android
I've enabled the Google Pay payment method from the Stripe dashboard. but still unable to see the GPay button on the stripe payment sheet.
Additionally, I've installed the Google Wallet app only.
Could you help me?

**Reply by nayanbabariya** [reply]
@captnCC @empiteranga @remonh87 Are there additional steps required to set up the `GPay` button on the stripe payment sheet?  because I am from India and GPay is not fully supported here.
I've done the Apple Pay integration as they have sandbox account testing and everything works fine for that.

**Reply by empiteranga** [reply]
@nayanbabariya No idea , but when im trying with live stripe keys and google pay test mode false , google pay button display , but I cant complete the payment need to verify the account , for verification need to run it with test mode , when in test mode google pay button doesn't display

**Reply by nayanbabariya** [reply]
@empiteranga Could you guide me about the GPay things? I am confused which is the app needed on the mobile device. GPay or Google Wallet app? plus, there is a link to enable test mode in the Google email => https://groups.google.com/g/googlepay-test-mode-stub-data

I've also joined that group, but not have an idea on how to register test cards on the wallet app.

**Reply by Nikunj-Flutter** [reply]
any update on the issue as i am still facing the same issue @nayanbabariya @empiteranga

**Comment by mihiravdevs** [comment]
Any solution?

**Comment by nayanbabariya** [comment]
@Nikunj-Flutter @mihiravdevs After a long time spent on this issue, I've a solution. You need the following things to see a Google Pay button on the `testEnv` Stripe sheet.

- The Device should have Activate VPN to the country where Google Pay is supported (Not the one that we Indians are using for UPI payment).
- Need a Google account that must be created in the country of your activated VPN
- The Google Wallet app should be installed and have 1 active card (However, in the test mode, your money will not be deducted 😄)
- You need to install and add 1 real card in Google Wallet only after connecting to a VPN. and same is applicable when you run your app.

**Comment by ABDELMAJIDRami** [comment]
The solution:

1. Download Google Wallet app on real android device not emulator
2. Setup a real debit card in the wallet app. (Don't worry you won't be charged)
3. add this code under /android/app/src/main/AndroidManifest.xml inside the `<application>...</application>` tag: 
```
<meta-data android:name="com.google.android.gms.wallet.api.enabled" android:value="true" />
```
4. Use this code in your flutter dart code (you can add more params):
```
await Stripe.instance.initPaymentSheet(
  paymentSheetParameters: SetupPaymentSheetParameters(
    paymentIntentClientSecret: clientSecret,
    merchantDisplayName: "Example Name",
    applePay: const PaymentSheetApplePay(merchantCountryCode: 'US'),
    googlePay: const PaymentSheetGooglePay(
      merchantCountryCode: 'US',  // country where your business is legally registered - not the customer’s location.
      currencyCode: 'USD',  // must match the one in your Stripe PaymentIntent created in the backend
      testEnv: true // remove in production as the default is false
    )
  )
);

await Stripe.instance.presentPaymentSheet();
```
5. Stop your app then run it again. Hot restart won't work.
6. You should now see G Pay option in the payment sheet with your card
7. Click on it - don't worry you won't be charged
8. You will prompted to Pay and you can select from many testing cards from google to simulate success, failure, 3ds, etc
